### ISSUE TITLE
Add support for Passport performance spans

### ISSUE THREAD

**Issue by olivernybroe** [getsentry/sentry-laravel#619]
Currently when using Laravel Passport we get a big black hole in the performance spans window, making it quite hard to know what is taking the performance there.

My theory is that this is due to the nature of how Passport works underneath the hood, but it would be awesome to fill the gaps here so we can improve the performance without going completely blind in when looking at the performance :) 
![image](https://user-images.githubusercontent.com/5870441/203530317-c91c0ccc-0166-4a6f-8f88-7eedbd283ee4.png)

**Comment by cleptric** [comment]
Thanks for reporting this, I'll take a look if we can improve this a bit 🙂

**Comment by stayallive** [comment]
Hehe, yeah this is an interesting problem, we can certainly try to take a look if we instrument more there but you can see that it get's "stuck" in the `authorize` method of the `AuthorizationController`, but that seems like a custom implementation since usually this lives in the Passport namespace, can you share that controller method and what it's doing or calling?

My suspician is that this has to do with the way the League OAuth2 server handles token where it encrypts them, and it taking a long time to get/parse the keys and generate the encrypted values, this is going to be very hard/impossible to profile without some hacks, we can't profile arbitrary PHP code being executed so we need a "PHP" way to inject our profiling using a decorator or other pattern which the OAuth2 server package is not really built for :)

Anyway long story short, this "delay" you are seeing has been a pain in the neck for me too and I have yet to pinpoint it, so trying to see if we can profile more is a very good excuse to look deeper into this!

**Comment by olivernybroe** [comment]
@stayallive glad to hear I am not the only one seeing this issue 😆 

You are right, I do have a custom implementation of it, but we don't change the `authorize` method :) 
https://gist.github.com/olivernybroe/63d1f6734a6f30bcb5e451f534883bfa

Hmm, that makes sense, I was hoping League OAuth has events we could listen into or telemetry we could attach ourself to or something like that 👀 

But thanks for wanting to take a deeper look into this 🤩

**Comment by stayallive** [comment]
Your method is executed (on the happy path), so you could maybe add a span there to see where in the flow it's getting there and how long the vendor code takes, I don't think your code takes any time but that could at least confirm the suspicion that it's not some other part of the application that is causing the slowdown.

Something like this: https://gist.github.com/stayallive/bb646ef80c0cb17252be091c3f276c60.

(the `traceWithSentry` is copied pastaad from #598 for the PoC)

**Comment by olivernybroe** [comment]
Hmm, interesting, that's of course a a good idea to create spans around the vendor code to reduce where it could be happening 🤔
